### DISCUSSION TITLE
capacitor-firebase app-check + Firebase JS SDK

### DISCUSSION THREAD

**Discussion by Klossek** [capawesome-team/capacitor-firebase#371] [Q&A] [answered]
1. I'm not entirely sure if I understand the purpose of the plugin correctly. Does it just add the native code to initialize the native app check providers?
2. Is there a way to use the JS Firebase API functions with activated app check also in IOS and Android apps?

**Comment by robingenz** [comment, marked as answer]
> I'm not entirely sure if I understand the purpose of the plugin correctly. Does it just add the native code to initialize the native app check providers?

Exactly

> Is there a way to use the JS Firebase API functions with activated app check also in IOS and Android apps?

Yes, please take a look at https://github.com/capawesome-team/capacitor-firebase/blob/main/packages/app-check/docs/firebase-js-sdk.md

**Reply by Klossek** [reply]
Hi Robin,

thanks a lot, I have tested it and it works on all platforms . I read the documentation link earlier but I was a bit overwhelmed and that`s why I didn't realise that this was exactly what I needed.

Maybe you could help me one more time with one related AppCheck issue. 
When I enable App Check I have a bit of a problem with image caching.My images are cached. With a hard coded src it works fine. But with appcheck enabled I need to use the firebase/storage getDownloadURL function to get the image src. Now foreach image a server request is performed and the images now appear with a delay. 
Could you recommend a library, or any solution to fix this? I think the images should be downloaded only once and then kept in local storage.
best regards 

Golo

**Reply by robingenz** [reply]
Sorry, I have never tested that specific use case before. Maybe someone on StackOverflow can answer your question.
